crates/anvil-polkadot/tests/it/time_machine.rs

```diff
@@ -0,0 +1,214 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use crate::utils::{assert_with_tolerance, response_result_success_inner, TestNode};
+use alloy_primitives::U256;
+use anvil_core::eth::EthRequest;
+use anvil_polkadot::config::{AnvilNodeConfig, SubstrateNodeConfig};
+use anvil_rpc::{
+    error::{ErrorCode, RpcError},
+    response::ResponseResult,
+};
+
+// Tests --------- EvmSetTime
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_evm_set_invalid_param() {
+    let anvil_node_config = AnvilNodeConfig::test_config();
+    let substrate_node_config = SubstrateNodeConfig::new(&anvil_node_config);
+    let mut node = TestNode::new(anvil_node_config, substrate_node_config).await.unwrap();
+    // Try to set the time too far ahead.
+    assert!(matches!(
+        node.eth_rpc(EthRequest::EvmSetTime(U256::from(u64::MAX))).await.unwrap(),
+        ResponseResult::Error(RpcError {
+            code: ErrorCode::InvalidParams,
+            message,
+            data: None
+        }) if message == "The timestamp is too big"
+    ));
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_evm_set_time_in_the_past() {
+    let anvil_node_config = AnvilNodeConfig::test_config();
+    let substrate_node_config = SubstrateNodeConfig::new(&anvil_node_config);
+    let mut node = TestNode::new(anvil_node_config, substrate_node_config).await.unwrap();
+
+    let _ = node.eth_rpc(EthRequest::Mine(None, None)).await.unwrap();
+    let first_hash = node.block_hash_by_number(1).await.unwrap();
+    let first_timestamp = node.get_decoded_timestamp(Some(first_hash)).await;
+
+    // Set the timestamp in the past
+    let new_timestamp = first_timestamp.saturating_div(1000).saturating_sub(1);
+    assert_eq!(
+        response_result_success_inner::<u64>(
+            node.eth_rpc(EthRequest::EvmSetTime(U256::from(new_timestamp))).await.unwrap()
+        )
+        .unwrap(),
+        0
+    );
+
+    let _ = node.eth_rpc(EthRequest::Mine(None, None)).await.unwrap();
+    let second_hash = node.block_hash_by_number(2).await.unwrap();
+    let second_timestamp = node.get_decoded_timestamp(Some(second_hash)).await;
+    assert_eq!(second_timestamp.saturating_sub(first_timestamp), 1);
```

Comment:
add a comment here saying that this does not work yet, otherwise the test is confusing